src/us/kbase/shock/client/BasicShockClient.java

```diff
@@ -406,14 +406,19 @@ public InputStream getFile(final ShockNode sn)
 	/**
 	 * Creates a node on the shock server containing a file.
 	 * @param file the file data.
+	 * @param fileLength the length of the file in bytes.
 	 * @param filename the name of the file.
 	 * @param format the format of the file, e.g. ASCII, UTF-8, JSON. Ignored
 	 * if null or whitespace only.
 	 * @return a shock node object.
 	 * @throws IOException if an IO problem occurs.
 	 * @throws ShockHttpException if the node could not be created.
 	 */
-	public ShockNode addNode(final InputStream file, final String filename, final String format)
+	public ShockNode addNode(
```

Comment:
Just a style question - when you add new params, you put them in the middle of the existing params?

Reply:
yep, doesn't really matter. Any code that uses this method won't compile either way, and the code will have to be update either way.

Reply:
Also the file size relates most closely to the file data, so I wanted them next to each other.